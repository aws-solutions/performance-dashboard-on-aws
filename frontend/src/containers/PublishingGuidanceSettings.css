.margin-top-0-important {
  margin-top: 0 !important;
}

.publishing-guidance p {
  margin: 0;
}
