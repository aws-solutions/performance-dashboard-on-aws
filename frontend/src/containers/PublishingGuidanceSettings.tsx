/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { useHistory } from "react-router-dom";
import { useSettings } from "../hooks";
import SettingsLayout from "../layouts/Settings";
import Button from "../components/Button";
import MarkdownRender from "../components/MarkdownRender";
import Spinner from "../components/Spinner";
import "./PublishingGuidanceSettings.css";
import { useTranslation } from "react-i18next";

function PublishingGuidanceSettings() {
    const history = useHistory();
    const { settings, loadingSettings } = useSettings();
    const { t } = useTranslation();

    const onEdit = () => {
        history.push("/admin/settings/publishingguidance/edit");
    };

    return (
        <SettingsLayout>
            <h1>{t("PublishingGuidance")}</h1>

            <p>{t("PublishingGuidanceDescription")}</p>

            {loadingSettings ? (
                <Spinner className="margin-top-3 text-center" label={t("LoadingSpinnerLabel")} />
            ) : (
                <>
                    <div className="grid-row margin-top-0-important">
                        <div className="grid-col flex-9">
                            <h2 className="text-bold">{t("AcknowledgeStatement")}</h2>
                        </div>
                        <div className="grid-col flex-3 text-right">
                            <Button className="margin-top-2" variant="outline" onClick={onEdit}>
                                {t("Edit")}
                            </Button>
                        </div>
                    </div>

                    <div className="grid-row margin-top-0-important">
                        <div className="grid-col flex-9">
                            <div className="publishing-guidance font-sans-lg">
                                <MarkdownRender source={settings.publishingGuidance} />
                            </div>
                        </div>
                        <div className="grid-col flex-3 text-right"></div>
                    </div>
                </>
            )}
        </SettingsLayout>
    );
}

export default PublishingGuidanceSettings;
