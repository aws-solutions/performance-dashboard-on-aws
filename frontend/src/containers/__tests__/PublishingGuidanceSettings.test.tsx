/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import PublishingGuidanceSettings from "../PublishingGuidanceSettings";

jest.mock("../../hooks");

test("renders the title", async () => {
    const { getByRole } = render(<PublishingGuidanceSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByRole("heading", { name: "Publishing guidance" })).toBeInTheDocument();
});

test("renders the description", async () => {
    const { getByText } = render(<PublishingGuidanceSettings />, {
        wrapper: MemoryRouter,
    });
    expect(
        getByText(
            "Publishing guidance is text that users must acknowledge before they " +
                "publish a dashboard. For example, use this text to remind them to check " +
                "for errors or mistakes, sensitive or confidential data, or guidance " +
                "specific to your organization.",
        ),
    ).toBeInTheDocument();
});

test("renders the acknowledgement statement header", async () => {
    const { getByText } = render(<PublishingGuidanceSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByText("Acknowledge statement")).toBeInTheDocument();
});

test("renders the acknowledgement statement", async () => {
    const { getByText } = render(<PublishingGuidanceSettings />, {
        wrapper: MemoryRouter,
    });
    expect(
        getByText("I acknowledge that I have reviewed the dashboard and it is ready to publish"),
    ).toBeInTheDocument();
});

test("renders a button to edit", async () => {
    const { getByRole } = render(<PublishingGuidanceSettings />, {
        wrapper: MemoryRouter,
    });
    const button = getByRole("button", { name: "Edit" });
    expect(button).toBeInTheDocument();
});

test("publishing guidance settings should match snapshot", async () => {
    const wrapper = render(<PublishingGuidanceSettings />, {
        wrapper: MemoryRouter,
    });
    expect(wrapper.container).toMatchSnapshot();
});
