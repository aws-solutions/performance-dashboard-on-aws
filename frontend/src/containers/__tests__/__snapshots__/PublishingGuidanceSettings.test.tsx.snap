// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`publishing guidance settings should match snapshot 1`] = `
<div>
  <div
    class="usa-section padding-top-3"
  >
    <div
      class="grid-container padding-x-0"
    >
      <div
        class="grid-row grid-gap"
      >
        <div
          class="desktop:grid-col-3"
        >
          <nav
            aria-label="Left navigation"
          >
            <ul
              class="usa-sidenav"
            >
              <li
                class="usa-sidenav__item"
              >
                <a
                  class="usa-current"
                  href="/admin/settings"
                >
                  Settings
                </a>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      aria-current="page"
                      class="usa-current"
                      href="/admin/settings/topicarea"
                    >
                      Topic areas
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/publishingguidance"
                    >
                      Publishing guidance
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/brandingandstyling"
                    >
                      Branding and styling
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/publishedsite"
                    >
                      Published site
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/dateformat"
                    >
                      Date and time format
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/adminsite"
                    >
                      Admin site
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="desktop:grid-col-9 usa-prose mobile-margin-top-2"
          id="main-content"
        >
          <h1>
            Publishing guidance
          </h1>
          <p>
            Publishing guidance is text that users must acknowledge before they publish a dashboard. For example, use this text to remind them to check for errors or mistakes, sensitive or confidential data, or guidance specific to your organization.
          </p>
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-9"
            >
              <h2
                class="text-bold"
              >
                Acknowledge statement
              </h2>
            </div>
            <div
              class="grid-col flex-3 text-right"
            >
              <button
                class="usa-button usa-button--outline margin-top-2"
              >
                Edit
              </button>
            </div>
          </div>
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-9"
            >
              <div
                class="publishing-guidance font-sans-lg"
              >
                <div
                  class="Markdown undefined"
                >
                  <p>
                    I acknowledge that I have reviewed the dashboard and it is ready to publish
                  </p>
                </div>
              </div>
            </div>
            <div
              class="grid-col flex-3 text-right"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
